addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_nodes.py

```diff
@@ -50,16 +51,17 @@ def gather_node(blender_object, export_settings):
     )
     node.translation, node.rotation, node.scale = __gather_trans_rot_scale(blender_object, export_settings)
 
-    if blender_object.type == 'LIGHT' and export_settings[gltf2_blender_export_keys.LIGHTS]:
-        correction_node = __get_correction_node(blender_object, export_settings)
-        correction_node.extensions = {"KHR_lights_punctual": node.extensions["KHR_lights_punctual"]}
-        del node.extensions["KHR_lights_punctual"]
-        node.children.append(correction_node)
-    if blender_object.type == 'CAMERA' and export_settings[gltf2_blender_export_keys.CAMERAS]:
-        correction_node = __get_correction_node(blender_object, export_settings)
-        correction_node.camera = node.camera
-        node.children.append(correction_node)
-    node.camera = None
+    if export_settings[gltf2_blender_export_keys.YUP]:
+        if blender_object.type == 'LIGHT' and export_settings[gltf2_blender_export_keys.LIGHTS]:
+            correction_node = __get_correction_node(blender_object, export_settings)
+            correction_node.extensions = {"KHR_lights_punctual": node.extensions["KHR_lights_punctual"]}
+            del node.extensions["KHR_lights_punctual"]
+            node.children.append(correction_node)
+        if blender_object.type == 'CAMERA' and export_settings[gltf2_blender_export_keys.CAMERAS]:
+            correction_node = __get_correction_node(blender_object, export_settings)
+            correction_node.camera = node.camera
+            node.children.append(correction_node)
+        node.camera = None
```

Comment:
It's a bit confusing.

Independently of Y-up or Z-up (or X-up?), I think it's universal that in the camera's coordinate system, +X is to the right, +Y is to the top, and +Z is behind the camera/screen.

If not using the YUP setting, then everything should just be left as it was, since blender's cameras follow that.

If I understand right, if using YUP, it basically rotates the coordinate system of each node around the X axis of each coordinate system 90° anti-clockwise (when seen with the X-axis pointing towards you). This has the side effect of rotating all meshes, which is undone by rotating the mesh data 90° clockwise around the X-axis. The camera gets rotated too, and since the camera isn't a mesh, the 90° clockwise correction can be either done by a correction node (the easy way) or by not applying the original YUP setting rotation to the camera node's coordinate system (the cleaner way, but a special case that need handling if the camera node has children or animations).

So if I understand right, the correction node should be applied if and only if using the YUP setting. I'd guess the same thing applies to the lights, but I haven't used them yet, and not sure how they should work.

Reply:
This has been taken pretty much exactly from [the original exporter](https://github.com/KhronosGroup/glTF-Blender-Exporter/blob/master/scripts/addons/io_scene_gltf2/gltf2_generate.py#L1663) and the check for the YUP-setting was forgotten (by me). You got it right!